include/triton/Dialect/TritonGPU/IR/LinearLayoutConversions.h

```diff
@@ -44,6 +44,64 @@ std::optional<LinearLayout>
 toLinearLayout(ArrayRef<int64_t> shape, Attribute layout,
                std::optional<int32_t> elemBitWidth = std::nullopt);
 
+// Given a linear layout with input dims and output dims containing a "block"
+// dimension, determines if the layout moves data across block boundaries.
+bool isCrossCTAConversion(const LinearLayout &layout);
+
+// Chooses a good shared layout for use when converting between the two given
+// register layouts.
+//
+// src and dst must have the following dimensions.
+//   input: [register, lane, warp, block]
+//   output: [dimi..] and [dimj..] where i and j are permutations over 0..N.
+//
+// The result has the following dimensions.
+//   input: [offset, block, iteration]
+//   output: matches src's output dims, so [dimi..].
+//
+// Note that a regular shared memory layout only has [offset, block] input dims.
+// The `iteration` dimension represents the fact that we can do a shmem transfer
+// in multiple stages to reduce the total amount of shmem needed.
+//
+// For example, imagine we have eight 32-bit registers per thread in src and
+// dst.  We could write all eight at once to shmem and then read them all back
+// in, but that requires a lot of shmem.  At the other extreme, we could write
+// one register to shmem and read it back in before writing the next one.  That
+// requires 8x less shmem but may prevent us from doing vectorized memory
+// accesses.
+//
+// As a compromise, this function splits the transfer so that we do one
+// vectorized load/store per iteration.  (In the example, we'd try to do two
+// stages of size 4 32-bit values each, assuming the src/dst layouts allow us to
+// vectorize the stores or loads as 4xi32.)
+LinearLayout chooseShemLayoutForRegToRegConversion(const LinearLayout &src,
```

Comment:
You mentioned over DM that this is unused but you're keeping it around because it might be useful later.  Personally I'd just keep it in a separate PR in that case, but if you do want to check it in, please add a comment indicating that this is known dead code but kept around for XYZ reason.

Reply:
Let me keep it in the separate PR